### PR TITLE
Handle exceptions

### PR DESCRIPTION
**Pull request by roback**
Raise `Twingly::Search::Error` instead of `RuntimeError` when something goes wrong.

Calls to the API can raise two types of exceptions `AuthError` and `ServerError`. Which one is raised is based on the response message from the server.

### CODE REVIEW

File: lib/twingly/search/error.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    class AuthError < Error
+    end
+
+    class ServerError < Error
+    end
+
+    class MissingAPIKeyError < Error
+      def initialize
+        super("No API key has been provided.")
```
Comment (walro) [inline lib/twingly/search/error.rb]: I think we can use `AuthError` and set this message.

Comment (roback) [inline lib/twingly/search/error.rb]: :+1:

Comment (roback) [inline lib/twingly/search/error.rb]: Done! bcdd76c89f6675c93a51713b77a07c348824ba34

File: spec/parser_spec.rb [hunk 2]
```diff
@@ -5,10 +5,45 @@
...
+
+    context "with a nonexistent api key result" do
+      let(:document) { Fixture.get(:nonexistent_api_key) }
+
+      it "should raise AuthError" do
+        expect { subject }.to raise_error(AuthError)
+      end
+    end
+
+    context "with a wrong api key result" do
```
Comment (walro) [inline spec/parser_spec.rb]: What we really mean with `wrong` is that they are using an API key for one of our other products, can we get that into the name (and fixture) without making it too wordy it would be great

**Comment by walro** [comment]
Minor comment, otherwise LGTM! Haven't done any real-life testing though.

File: spec/parser_spec.rb (hunk 2 above)
Comment (roback) [inline spec/parser_spec.rb]: `unauthorized` maby?

Comment (walro) [inline spec/parser_spec.rb]: yeah. that's better
